app/uk/gov/hmrc/agentsubscription/controllers/SubscriptionJourneyController.scala

```diff
@@ -84,20 +84,21 @@ class SubscriptionJourneyController @Inject() (
 
   // Find the SJR then overwrite the authProviderId, businessDetails and cleanCredsAuthProviderId. This allows the journey to progress when
   // the agent uses a different cred to continue the journey after some days/weeks and during that time some details may have changed e.g. address.
-  def handleConflict(sjr: SubscriptionJourneyRecord)(implicit ec: ExecutionContext): Future[Result] = {
-    val utr = sjr.businessDetails.utr
+  def handleConflict(sjr: SubscriptionJourneyRecord)(implicit ec: ExecutionContext): Future[Result] =
     for {
-      optExistingSjr <- subscriptionJourneyRepository.findByUtr(utr)
-      existingSjr    <- optExistingSjr.fold[Future[SubscriptionJourneyRecord]](logUTRError(sjr).toFailure)(_.toFuture)
-      updatedSjr = existingSjr.copy(
-                     authProviderId = sjr.authProviderId,
-                     businessDetails = sjr.businessDetails,
-                     cleanCredsAuthProviderId = sjr.cleanCredsAuthProviderId
-                   )
-      _      <- subscriptionJourneyRepository.updateOnUtr(utr, updatedSjr)
-      result <- Ok(toJson(updatedSjr)).toFuture
+      modifiedRecord <- subscriptionJourneyRepository.updateOnUtr(
+                          sjr.businessDetails.utr,
+                          Json.obj(
+                            "authProviderId"           -> sjr.authProviderId,
+                            "businessDetails"          -> sjr.businessDetails,
```

Comment:
businessDetails contains fields that need to be encrypted but it looks like you're relying on the implicit format here which is plaintext. The database formats won't apply here.
I'm also not quite sure if the rewrite here is necessary as just assuming all existing data is encrypted and correctly encrypting the utr before filtering should ensure it all works.

Reply:
thanks Val, we are agreed we can rollback this commit and work out how to approach updateOne with encryption as a pattern for the future.